README.md

```diff
@@ -48,7 +48,7 @@ app.use(cors());
  * CORS middleware
  *
  * @param {Object} [options]
- *  - {String|Function(ctx)} origin `Access-Control-Allow-Origin`, default is request Origin header
+ *  - {String|Function(ctx)|Promise} origin `Access-Control-Allow-Origin`, default is request Origin header
```

Comment:
`{String|Promise<string>}` may be better

Reply:
But that would be a breaking change wouldn't it? It supports functions as is.

Reply:
`promise` is still function, right? aren't you upgrade it from synchronous normal function to `promise`?